hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/RewardCalculator.java

```diff
@@ -73,13 +80,39 @@ boolean isWithinRange(final long stakePeriodStart, final long todayNumber) {
 		return stakePeriodStart > -1 && stakePeriodStart < todayNumber - 1;
 	}
 
-	long computeReward(final MerkleAccount account, final long stakedNode, final long todayNumber) {
+	long computeReward(final MerkleAccount account, final long stakedNode, final long todayNumber,
+			final long stakePeriodStart) {
 		final var stakedNodeAccount = stakingInfo.get().get(EntityNum.fromLong(stakedNode));
```

Comment:
Here, I'd also encapsulate the staking info `MerkleMap` in a `StakingInfoManager` component. 

(E.g. we might want to improve performance by only calling `getForModify()` once-per-round for each node's `StakingInfo`. The `StakingInfoManager` could accomplish this by keeping the mutable `StakingInfo`'s for the round in a `StakingInfo[]`, which it only refreshes when it sees the `Supplier<MerkleMap>` returns a different instance.)

Reply:
Made changes